src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,180 @@
+import classNames from 'classnames';
+import {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import * as todoService from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../../TodosContext';
+import { ErrorContext } from '../../ErrorContext';
+import { Error } from '../../types/ErrorMessage';
+
+const ERROR_DELAY = 3000;
+
+type Props = {
+  todo: Todo;
+  loadingIds: number[];
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  loadingIds,
+}) => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(todo.title);
+  const { setErrorMessage } = useContext(ErrorContext);
+  const refInput = useRef<HTMLInputElement>(null);
+
+  const deleteTodo = (todoId: number) => {
+    setIsLoading(true);
+
+    todoService.deleteTodo(todoId)
+      .then(() => {
+        setIsLoading(false);
+        const currentTodos = [...todos];
+
+        setTodos(currentTodos.filter(t => t.id !== todoId));
+      })
+      .catch((error) => {
+        setErrorMessage(Error.DELETE);
+        setIsLoading(false);
+
+        setTimeout(() => setErrorMessage(''), ERROR_DELAY);
+
+        throw error;
+      });
+  };
+
+  const updateTodo = (uTodo: Todo): Promise<void> => {
+    return todoService.updateTodo(uTodo)
+      .then(updatedTodo => {
+        const currentTodos = [...todos];
+        const index = currentTodos.findIndex(t => t.id === todo.id);
+
+        currentTodos.splice(index, 1, updatedTodo);
+        setTodos(currentTodos);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        setIsLoading(false);
+        setErrorMessage(Error.UPDATE_TODO);
+
+        setTimeout(() => setErrorMessage(''), ERROR_DELAY);
+        throw error;
+      });
+  };
+
+  const handleCompleteTodo = (uTodo: Todo) => {
```

Comment:
redundant wrapper, just move the logic of `updateTodo` here

Reply:
> redundant wrapper, just move the logic of `updateTodo` here

I use `updateTodo` in `handleCompleteTodo` and in `handleOnBlur`